pymarlin/core/trainer.py

```diff
@@ -157,15 +187,20 @@ def train(self):
             self.module.on_end_train_epoch(self.global_steps_finished, *all_outputs)
             self.stats.log_long_stats(self.global_steps_finished)
             self.last_epoch = epoch
-            self.save_checkpoint()
+            self.save_checkpoint() 
+
+        throughput =  (self.args.train_batch_size * (self.global_steps_finished - self.estimated_global_steps_per_epoch)) / (time.time() - start_train_stable_time)
+        self.stats.update("throughput", throughput, frequent=True)
+        run.log("throughput", throughput)
 
         # Checkpoint one final time at the end of training and save model
         self.save_checkpoint(force=True)
         self.save_model_checkpoint()
 
         self.module.on_end_train(self.global_steps_finished)
         self.stats.log_long_stats(self.global_steps_finished)
-        self.logger.info("Finished training .. ")
+        self.logger.info("Throughput for the run is: {} samples/s".format(round(throughput, 3)))
```

Comment:
unnecessary. Override `Trainer` and `train()` if these logs are needed for the usecase. This is a core class and should be changed only when absolutely necessary.

Reply:
stdout Writer in stats package already writes all the stats to stdout. Use that